content/docs/reference/buildpack-api.md

```diff
@@ -144,6 +137,92 @@ version = "1.0"
 id = "io.buildpacks.stacks.bionic"
 ```
 
+### Schema
+
+The schema is as follows:
+   
+- **`api`** _(string, required, current: `0.2`)_\
+    The Buildpack API version the buildpack adheres to. Used to ensure [compatibility](#api-compatibility) against
+    the [lifecycle][lifecycle].
+
+    > Not to be confused with Cloud Foundry or Heroku buildpack versions. This version pertains to the interface
+    > between the [buildpack][buildpack] and the [lifecycle][lifecycle] of Cloud Native Buildpacks.
+
+- **`buildpack`** _(required)_\
+    Information about the buildpack.
+
+    - **`id`** _(string, required)_\
+    A globally unique identifier.
+
+    - **`version`** _(string, required)_\
+    The version of the buildpack.
+    
+    - **`name`** _(string, required)_\
+    Human readable name.
+
+    - **`clear-env`** _(boolean, optional, default: `false`)_\
+    Clears user-defined environment variables when `true` on executions of `bin/detect` and `bin/build`.
+
+- **`stacks`** _(list, optional)_\
+    A list of stacks supported by the buildpack.
+    _If omitted, `order` list must be present. Cannot be used in conjunction with `order` list._
+
+    - **`id`** _(string, required)_\
+    The id of the supported stack.
+
+    - **`mixins`** _(string list, required)_\
+    A list of mixins required on the stack images.
+
+- **`order`** _(list, optional)_\
+  A list of buildpack groups. This list determines the order in which groups of buildpacks
+  will be tested during detection. For the purpose of creating a [meta-buildpack][meta-buildpack]. _If omitted, `stacks` list must be present. Cannot be used in conjunction with `stacks` list._
+  
+    - **`group`** _(list, required)_\
+    A set of buildpack references.
+
+        - **`id`** _(string, required)_\
+          The identifier of a buildpack from the configuration's top-level `buildpacks` list. Buildpacks with the same ID may
+          appear in multiple groups at once but never in the same group.
+    
+        - **`version`** _(string, required)_\
+          The version of the buildpack being referred to.
+    
+        - **`optional`** _(boolean, optional, default: `false`)_\
+          Whether or not this buildpack is optional during detection.
+ 
+- **`metadata`** _(any, optional)_\
+    Arbitrary data for buildpack.
+ 
+
+## API Compatibility
+
+API compatibility is determined using the logic:
+
+`SUBJECT` is the Buildpack API version of the **buildpack**.\
```

Comment:
Could we use `BP` and `LC`, or `BUILDPACK` and `LIFECYCLE` here instead? I found it difficult to continuously translate as I read.

Reply:
I agree that the names used are not optimal but not sure the proposed names would fit any better. I would be concerned that it could be construed as the version of the buildpack or the lifecycle instead of the _buildpack api version_ of the buildpack or lifecycle.

Reply:
Could we borrow some language from https://semver.org/? For example:

> Precedence is determined by the first difference when comparing each of these identifiers from left to right as follows: Major, minor, and patch versions are always compared numerically. Example: 1.0.0 < 2.0.0 < 2.1.0 < 2.1.1. When major, minor, and patch are equal, a pre-release version has lower precedence than a normal version. Example: 1.0.0-alpha < 1.0.0. Precedence for two pre-release versions with the same major, minor, and patch version MUST be determined by comparing each dot separated identifier from left to right until a difference is found as follows: identifiers consisting of only digits are compared numerically and identifiers with letters or hyphens are compared lexically in ASCII sort order. Numeric identifiers always have lower precedence than non-numeric identifiers.

Reply:
IIRC, semver compatibility is based on a simple comparison based (without considering [constraints](https://jubianchi.github.io/semver-check/#/)) on the precedence you quoted. Our compatibility is slightly more complex to describe.

#### By examples:

In semver as per [MINOR](https://semver.org/#spec-item-7): 1.2.0 IS compatible with 1.3.0. The inverse is also compatible. 1.3.0 IS compatible with 1.2.0.

In our API versions as per [spec](https://github.com/buildpack/spec/blob/master/buildpack.md#buildpacktoml-toml): A buildpack with Buildpack API 1.2 IS compatible with lifecycle with Buildpack API 1.3. The inverse is NOT compatible. A buildpack with Buildpack API 1.3 IS NOT compatible with lifecycle with Buildpack API 1.2.

Reply:
Updated. Let me know what you think.